HOPE.py

```diff
@@ -120,6 +125,12 @@
                 labels2d = labels2d.float()
                 labels3d = labels3d.float()
 
+                if use_cuda and torch.cuda.is_available():
+                    inputs = inputs.float().cuda(device=args.gpu_number[0])
+                    labels2d = labels2d.float().cuda(device=args.gpu_number[0])
+                    labels3d = labels3d.float().cuda(device=args.gpu_number[0])
+
+
```

Comment:
Almost the same as the Graph.py note.  Here the cuda branch was added after you already cast the variables as floats.   So this will be slow and then do the operation quickly as opposed to the last one that would do it quickly and then do it again slowly.